File: app/src/main/java/me/cargoapp/cargo/messaging/MessagingQueue.java
package me.cargoapp.cargo.messaging;

import java.util.ArrayList;

/**
 * Created by Marvin on 10/05/2017.
 */

public class MessagingQueue {
    static ArrayList<MessagingNotificationParser.NotificationParserResult> _messagesQueue = new ArrayList<MessagingNotificationParser.NotificationParserResult>();

    public static void add (MessagingNotificationParser.NotificationParserResult result) {
        _messagesQueue.add(result);
    }

    public static boolean isFilled () {
        return _messagesQueue.size() != 0;
    }

    public static MessagingNotificationParser.NotificationParserResult get () {
        return _messagesQueue.remove(0);
    }
}

File: app/src/main/java/me/cargoapp/cargo/messaging/MessagingQueue.kt
package me.cargoapp.cargo.messaging

import java.util.*

/**
 * Created by Marvin on 10/05/2017.
 */


object MessagingQueue {
    private var _messagesQueue = ArrayList<MessagingNotificationParser.NotificationParserResult>()

    fun add(result: MessagingNotificationParser.NotificationParserResult) {
        _messagesQueue.add(result)
    }

    val isFilled: Boolean
        get() = _messagesQueue.size != 0

    fun get(): MessagingNotificationParser.NotificationParserResult {
        return _messagesQueue.removeAt(0)
    }
}
